pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -106,38 +102,57 @@ func markRoutingState(
 	}
 }
 
-// addRouteLabel appends the route label to the list of labels if needed
-// or removes the label if routeName is nil.
-func addRouteLabel(acc kmeta.Accessor, routeName string, diffLabels map[string]interface{}) error {
-	if routeName == "" { // remove the label
-		if acc.GetLabels()[serving.RouteLabelKey] != "" {
-			diffLabels[serving.RouteLabelKey] = nil
+// addRouteAnnotation appends the route annotation to the list of labels if needed
+// or removes the annotation if routeName is nil.
```

Comment:
what does return value signify?
Also the name is strange given that it also _removes_ items from the annotation list, or even the annotation itself.

Reply:
Changed the name.
Added a comment for the return value, it is a bit odd: This returns true if the annotation was added or removed entirely, this is used to signify a state change from active/not-active.